x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/apikey/QueryApiKeyResponse.java

```diff
@@ -50,12 +57,15 @@ public static QueryApiKeyResponse emptyResponse() {
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject()
+            .field("total", total)
+            .field("count", foundApiKeysInfo.length)
             .array("api_keys", (Object[]) foundApiKeysInfo);
```

Comment:
I didn't add a `sort` field here in the response because adding it would make the response look less pleasing, i.e. it would be
```
{ "api_key": [ {"api_key":{...}, "sort":[...]}, {} ] }
```
instead of just
{ "api_key": [ {...}, {...} ] }

When `sort` is not required, the more nested version looks even more unnecessary. Also, unlike regular search response, all sort values are present in the response (`api_key`) already since there will be no filtering. So the information can be readliy parsed from each api_key item.

Reply:
I realised there is one minor caveat of not showing sort values: They can have different formats than the original value. For example, `creation_time` can be formatted to be more human readable. So instead of
```
"sort": [ 1626237059434 ]
```
it will be the following more human readable format
```
"sort" : [ "2021-07-14T04:30:59.434Z" ]
```
Note this difference does not seem to affect how `search_after` works. That is, you can pass specify either `"search_after": [1626237059434]` or `"search_after": ["2021-07-14T04:30:59.434Z"]` and they both work.

In general, while the human readable format of sort values is nice to have, I don't think it is worth to complicate the response especially since `search_after` does not get impacted.